bears/general/IndentationBear.py

```diff
@@ -99,6 +105,16 @@ def run(self,
             new_file.append(insert*indent_levels[line_nr] + line
                             if line is not "\n" else "\n")
 
+        for _range, indent in absolute_indent_levels:
```

Comment:
your run functions is getting *pretty* large now. You are doing a lot of things sequentially. Maybe you can it into a couple of different (_private) functions?